qiskit/primitives/containers/data_bin.py

```diff
@@ -15,78 +15,78 @@
 """
 from __future__ import annotations
 
-from collections.abc import Iterable
-from dataclasses import make_dataclass
+from typing import Any, Iterable
 
 
 class DataBinMeta(type):
-    """Metaclass for :class:`DataBin` that adds the shape to the type name.
+    """This metaclass no longer has a purpose and will be removed."""
 
-    This is so that the class has a custom repr with DataBin<*shape> notation.
-    """
 
-    def __repr__(cls):
-        name = cls.__name__
-        if cls._SHAPE is None:
-            return name
-        shape = ",".join(map(str, cls._SHAPE))
-        return f"{name}<{shape}>"
+class DataBin(metaclass=DataBinMeta):
+    """Namespace for storing data.
 
+    .. code-block:: python
 
-class DataBin(metaclass=DataBinMeta):
-    """Base class for data bin containers.
+        data = DataBin(
+            alpha=BitArray.from_bitstrings(["0010"]),
+            beta=np.array([1.2])
+        )
+
+        print("alpha data:", data.alpha)
+        print("beta data:", data.beta)
 
-    Subclasses are typically made via :class:`~make_data_bin`, which is a specialization of
-    :class:`make_dataclass`.
     """
 
     _RESTRICTED_NAMES = ("_RESTRICTED_NAMES", "_SHAPE", "_FIELDS", "_FIELD_TYPES")
-    _SHAPE: tuple[int, ...] | None = None
-    _FIELDS: tuple[str, ...] = ()
-    """The fields allowed in this data bin."""
-    _FIELD_TYPES: tuple[type, ...] = ()
-    """The types of each field."""
+
+    def __init__(self, **kwargs):
+        self.__dict__.update(kwargs)
+
+    def __setattr__(self, key, value):
+        raise NotImplementedError
 
     def __len__(self):
-        return len(self._FIELDS)
+        return len(self.__dict__)
```

Comment:
Is this safe? there won't be any extra fields in the classes dict format (eg if we added a shape /ndim to it which I would find very useful in majority of the post-processing i do with these classes)

Reply:
See a different answer.